contracts/contracts/strategies/LidoWithdrawalStrategy.sol

```diff
@@ -0,0 +1,307 @@
+        require(stETHStart > 0, "No stETH to withdraw");
+
+        uint256 numWithdrawals = (stETHStart / MaxWithdrawalAmount) + 1;
+        uint256[] memory amounts = new uint256[](numWithdrawals);
+
+        uint256 stETHRemaining = stETHStart;
+        uint256 i = 0;
+        while (stETHRemaining > MaxWithdrawalAmount) {
+            amounts[i++] = MaxWithdrawalAmount;
+            stETHRemaining -= MaxWithdrawalAmount;
+        }
+        amounts[i] = stETHRemaining;
+
+        uint256[] memory requestIds = withdrawalQueue.requestWithdrawals(
+            amounts,
+            address(this)
+        );
+
+        emit WithdrawalRequests(requestIds, amounts);
+
+        // Is there any stETH left except 1 wei from each request?
+        require(
+            stETH.balanceOf(address(this)) <= numWithdrawals,
+            "Not all stEth in withdraw queue"
+        );
+        outstandingWithdrawals += stETHStart; // Single set for gas reasons
+
+        // This strategy claims to support WETH, so it is possible for
+        // the vault to transfer WETH to it. This returns any deposited WETH
+        // to the vault so that it is not lost for balance tracking purposes.
+        uint256 wethBalance = weth.balanceOf(address(this));
+        if (wethBalance > 0) {
+            // slither-disable-next-line unchecked-transfer
+            weth.transfer(vaultAddress, wethBalance);
+        }
+
+        emit Deposit(address(stETH), address(withdrawalQueue), stETHStart);
+    }
+
+    /**
+     * @notice Withdraw an asset from the underlying platform
+     * @param _recipient Address to receive withdrawn assets
+     * @param _asset Address of the asset to withdraw
+     * @param _amount Amount of assets to withdraw
+     */
+    function withdraw(
+        // solhint-disable-next-line no-unused-vars
+        address _recipient,
+        // solhint-disable-next-line no-unused-vars
+        address _asset,
+        // solhint-disable-next-line no-unused-vars
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        // Does nothing - all withdrawals need to be called manually using the
+        // Strategist calling claimWithdrawals
+        revert("use claimWithdrawals()");
+    }
+
+    /**
+     * @notice Claim previously requested withdrawals that have now finalized.
+     * Called by the Strategist.
+     * @param _requestIds Array of withdrawal request identifiers
+     * @param expectedAmount Total amount of ETH expect to be withdrawn
+     */
+    function claimWithdrawals(
+        uint256[] memory _requestIds,
+        uint256 expectedAmount
+    ) external nonReentrant {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        uint256 startingBalance = payable(address(this)).balance;
+        uint256 lastIndex = withdrawalQueue.getLastCheckpointIndex();
+        uint256[] memory hintIds = withdrawalQueue.findCheckpointHints(
```

Comment:
Have you considered doing this off-chain and passing the hint to the function?

Reply:
I like the thinking as it'll save gas on the `getLastCheckpointIndex` and `findCheckpointHints` calls. Looking at our stETH ARM, it'll save around 16k in gas.

![059e3103](https://github.com/OriginProtocol/origin-dollar/assets/6491112/a87c577f-57de-4498-95fe-ec762e5a4dd5)

Given this is one-off use, I think can leave the `getLastCheckpointIndex` and `findCheckpointHints` calls in the contract. gas is relatively cheap at the moment :-)